### PR TITLE
Show how to install cabal-head via GHCup

### PR DESCRIPTION
**Pull request by ffaf1**
---
Please include the following checklist in your PR:

* [x] The documentation has been updated, if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ _Getting unreleased versions of `cabal-install`_: gives you a chance to try out
 Currently, we only provide binaries for `x86_64` platforms.
 
 1. _[GitHub preview release built from the tip of the `master` branch](https://github.com/haskell/cabal/releases/tag/cabal-head)_:
```
Comment (ulysses4ever) [inline README.md:36]:
```suggestion
1. _GitHub preview release built from the tip of the `master` branch_: [download from GitHub(https://github.com/haskell/cabal/releases/tag/cabal-head) or use this GHCup command:
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,11 @@ _Getting unreleased versions of `cabal-install`_: gives you a chance to try out
...
+   ```
+   # Or install it via GHCup
```
Comment (ulysses4ever) [inline README.md:38]:
```suggestion
```

**Review by ulysses4ever — APPROVED** [review]
Grammatically, that "Or" doesn't quite make sense to me. I added a suggestion to add some text and move there your #-comment.

File: README.md (hunk 1 above)
Comment (ffaf1) [reply README.md]: I have added a “to install” after “GHCup command” for clarity.

**Review by Kleidukos — APPROVED** [review]

**Comment by mouse07410** [comment]
Any reason why AARCH64 binaries aren't available, e.g., for MacOS?

**Comment by ulysses4ever** [comment]
Actually, wet should add a comment that `Linux` can be replaced with `Windows` or `MacOS`.

**Comment by ulysses4ever** [comment]
@mouse07410 because these are GitHub prereleases, and GitHub doesn't provide AARCH64 runners. Releases, on the other hand, are built on GitLab where we have "our own" runners, including AARCH64. If you think you could implement a prerelease workflow that would employ GirLab, I think, everyone will be very grateful.

**Review by hasufell — APPROVED** [review]

**Comment by ulysses4ever** [comment]
@ffaf1 merge label?

**Comment by hasufell** [comment]
> and GitHub doesn't provide AARCH64 runners

It does. @angerman and me have set up and developed CI that works with privately hosted runners on aarch64 macOS.

We have repeatedly advertised this capability and I have repeatedly offered to migrate all Cabal CI to Github, but cabal devs have always raised concerns.

So... this is up to cabal devs. We can enable and expose these runners if need be.

**Comment by gbaz** [comment]
> We can enable and expose these runners if need be.

I think we would all love that for the CI pipeline, and there is no objection!

The disagreements were over what would happen with the release pipeline, but we need not let that get in the way of making our CI system better :-)

**Comment by ulysses4ever** [comment]
By all means. Just want to make sure that everyone is on the same page here: the prerelease workflow is currently completely independent of release process, and I'm not aware of any plans to converge them.

File: README.md [hunk 3]
```diff
@@ -33,7 +33,14 @@ Ways to get the `cabal-install` binary
...
 Currently, we only provide binaries for `x86_64` platforms.
 
-1. _[GitHub preview release built from the tip of the `master` branch](https://github.com/haskell/cabal/releases/tag/cabal-head)_:
+1.  _GitHub preview release built from the tip of the `master` branch_: [download from GitHub](https://github.com/haskell/cabal/releases/tag/cabal-head) or use this GHCup command to install:
+
+    ```
+    ghcup install cabal -u https://github.com/haskell/cabal/releases/download/cabal-head/cabal-head-Linux-x86_64.tar.gz head
+    ```
+
+    Replace "Linux" with "Windows" or "MacOS" as appropriate.
```
Comment (ulysses4ever) [inline README.md:42]:
```suggestion
    Replace "Linux" with "Windows" or "macOS" as appropriate.
```

This is weird but that's how it's spelled by GitHub somehow...

**Comment by ulysses4ever** [comment]
@mergify rebase

**Pull request merged**